### DISCUSSION TITLE
How to setup AVA to work with both javascript and typescript files?

### DISCUSSION THREAD

**Discussion by maxfarseer** [avajs/ava#3324] [Q&A]
Hello, I have a codebase written in javascript and I would like to migrate it to typescript.

I was able to setup ava to run tests for `.ts` files copying setup from this example ([ts receipe page](https://github.com/avajs/ava/blob/main/docs/recipes/typescript.md), [playground](https://stackblitz.com/github/avajs/ava/tree/main/examples/typescript-basic?file=source%2Ftest.ts&terminal=test&view=editor)). Ts works!

If I will add `.js` files to scenario, I had an error:
```
import test from "ava";
  ^^^^^^

  SyntaxError: Cannot use import statement outside a module
```

<img width="591" alt="Screenshot 2024-04-18 at 17 41 12" src="https://github.com/avajs/ava/assets/1026825/c624f54f-98f2-43b1-8e3d-21d8a4196ac0">

Here is the [my example repo](https://github.com/maxfarseer/ts-js-ava) with representation of the issue:

inside `package.json`

```
"files": [
      "**/*.ava.ts",
      "**/*.ava.js" // <!-- remove this line and ts tests are pass.
    ]
```

Any tips to make both `js` and `ts` work together?

Thank you,
Max.

**Comment by novemberborn** [comment]
You haven't told Node.js to treat `.js` files as ES modules, so you cannot use `import`. This isn't an AVA problem as such.
